web/src/engine/websites/Mangaz.ts

```diff
@@ -59,39 +68,31 @@ export default class extends DecoratableMangaScraper {
 
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
         const script = `
-            new Promise(async (resolve,reject) => {
-                let g = JCOMI.namespace("JCOMI.document")
-                let b = g.getDoc()
-                let img = g.getImages().map(ele => g.getLocationDir('enc') + ele.file + "?vw=" + encodeURIComponent(JCOMI.namespace("JCOMI.config").getVersion()))
-                resolve({img:img,b:b});
+            new Promise(resolve => {
+                const jNamespace = JCOMI.namespace("JCOMI.document");
+                const jDocument = jNamespace.getDoc();
+                const enc = jDocument.Location.enc ? 'enc' : 'anne';
+                const imgs = jNamespace.getOrders().map(ele => {
+                    const img = jNamespace.getLocationDirAnne(enc) + ele.name + "?" + jDocument.verkey;
+                    return { img, scrambleData : ele.scramble };
+                });
+                resolve(imgs);
             });
         `;
-        const request = new Request(new URL(chapter.Identifier, this.URI).href);
-        const data = await FetchWindowScript<ImgObj>(request, script);
-        return data.img.map(ele => new Page(this, chapter, new URL(ele), data.b.Enc));
+        const request = new Request(new URL(chapter.Identifier, this.URI));
+        const data = await FetchWindowScript<ImageData[]>(request, script, 2500);
+        return data.map(image => new Page(this, chapter, new URL(image.img, this.URI), image.scrambleData));
     }
 
     public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
-        const data = await Common.FetchImageAjax.call(this, page, priority, signal);
-        const encrypted = await new Response(data).arrayBuffer();
-        const iv = Buffer.from(window.btoa(page.Parameters.iv as string), 'base64');
-        const key = Buffer.from(page.Parameters.key as string, 'utf-8');
-
-        const secretKey = await crypto.subtle.importKey(
-            'raw',
-            key,
-            {
-                name: 'AES-CBC',
-                length: 128
-            }, true, ['encrypt', 'decrypt']);
-
-        let decrypted = await crypto.subtle.decrypt({
-            name: 'AES-CBC',
-            iv: iv,
-        }, secretKey, encrypted,);
-
-        const sdecrypted = new TextDecoder('utf-8').decode(decrypted);
-        decrypted = Uint8Array.from(window.atob(sdecrypted), char => char.charCodeAt(0));
-        return await Common.GetTypedData(decrypted);
+        const blob = await Common.FetchImageAjax.call(this, page, priority, signal);
+        const scrambleData = page.Parameters as ScrambleData;
```

Comment:
Optional typing is now available for page parameters and should be used when possible
See: https://github.com/manga-download/haruneko/commit/abb147fdbf7079781a5e4e44f6c83ccf3f1b3aa6

Reply:
Nice !